### PR TITLE
Conditional validation when parameter is not required. [patch]

### PR DESCRIPTION
**Pull request by isoos**
Applying #159 on the generated packages. We could probably gain an extra line per check if e.g. `validateStringLength` could take an optional parameter `isRequired` and do the null check inside, maybe if we need to add another check for some reason...

### CODE REVIEW

File: generated/aws_elasticache_api/lib/elasticache-2015-02-02.dart [hunk 1]
```diff
@@ -3316,6 +3321,10 @@ class ElastiCache {
     bool showNodeLevelUpdateStatus,
     List<String> updateActionStatus,
   }) async {
+    if (cacheClusterIds != null) {}
+    if (replicationGroupIds != null) {}
+    if (serviceUpdateStatus != null) {}
+    if (updateActionStatus != null) {}
     final $request = <String, dynamic>{
```
Comment (Schwusch) [inline generated/aws_elasticache_api/lib/elasticache-2015-02-02.dart:3328]: This didn't go as expected..?

Comment (isoos) [reply generated/aws_elasticache_api/lib/elasticache-2015-02-02.dart:3328]: Good catch, maybe we should fix this before commiting the change... Are you up for it?

Comment (Schwusch) [reply generated/aws_elasticache_api/lib/elasticache-2015-02-02.dart:3328]: Yeah I'll take a shot at it

Comment (Schwusch) [reply generated/aws_elasticache_api/lib/elasticache-2015-02-02.dart:3328]: https://github.com/agilord/aws_client/pull/162

Comment (isoos) [reply generated/aws_elasticache_api/lib/elasticache-2015-02-02.dart:3328]: Thank you for the quick turnaround! I'm running the codegen again for both PRs.

**Pull request merged**